hub/store/dynamic_tensor.py

```diff
@@ -244,39 +244,56 @@ def check_value_shape(self, value, slice_):
                     raise ValueShapeError(expected_value_shape, value.shape)
         return value
 
+    def get_shape_samples(self, samples):
+        """Gets full shape of dynamic_tensor(s)"""
+        if isinstance(samples, int):
+            shape, shape_offset = [], 0
+            for i in range(1, len(self.shape)):
+                if self.shape[i] is not None:
+                    current = self.shape[i]
+                else:
+                    current = self._dynamic_tensor[samples][shape_offset]
+                    shape_offset += 1
+                shape.append(current)
+            return shape
+        elif isinstance(samples, slice):
+            start = samples.start if samples.start is not None else 0
+            stop = samples.stop if samples.stop is not None else self.shape[0]
+            shapes = [self.get_shape_samples(i) for i in range(start, stop)]
```

Comment:
Why are you looping over, self._dynamic_tensor[samples] should give you all shapes for all samples.

Reply:
But the shapes of all dynamic tensors are different right? could be 100, 10 for first and 20, 1000 for the next. How would  self._dynamic_tensor[samples]  help in that case?